doc/src/arch/reference.rst

```diff
@@ -28,10 +28,18 @@ The architecture tag contains the following tags:
 
 Recognized BLIF Models (<models>)
 ---------------------------------
-The ``<models>`` tag contains ``<model name="string">`` tags.
+The ``<models>`` tag contains ``<model name="string" type="string">`` tags.
 Each ``<model>`` tag describes the BLIF ``.subckt`` model names that are accepted by the FPGA architecture.
 The name of the model must match the corresponding name of the BLIF model.
 
+The type of the model is not strictly required and it can assume the following values:
+
+* normal (default)
+* special
```

Comment:
Rather than having a type, maybe an attribute like `never_prune="true"`?

Reply:
The main reason to have a type was that it might get expanded in the future to allow some other kind of operations on different model types, also to leave this as generic as possible.

Having a `never_prune` attribute might be too over-specific from an architecture description point of view.

Reply:
But having it as a type is also never particularly generic.  A "type" works well if each addition type is very seperate.  But if each type enables some features but not others, attribute flags allow for better flexiblity.  So in this case, "special" is non-descriptive name, and the attribute name "type" is also not descriptive.

I'm open to other possibilities, but `type="special"` is not a good solution here.

Reply:
I see your point and I agree. I'll get the changes in and transition to use attribute flags instead of model types.